### ISSUE TITLE
Comparison to org-brain, org-wiki, org-roam, org-zettelkasten or other zettelkasten tools?

### ISSUE THREAD

**Issue by michaelsjackson** [EFLS/zetteldeft#56]
How would zetteldeft compare to org-brain or org-wiki or org-roam or other similar zettelkasten tools?

Currently I am using org-brain together with obvz mostly. But planning using org-wiki in parallel as well, as they do not disturb each other, and both can work on same org-files, but using kind of different "tunnels".

Now, I guess even zetteldeft can be added to this arsenal, bringing a third "tunnel" on same org-file notes collection.  What do others think? Any comments are welcome, thanks in advance.

I did not test zetteldeft yet.

**Comment by EFLS** [comment]
> Now, I guess even zetteldeft can be added to this arsenal, bringing a third "tunnel" on same org-file notes collection.

I don't think it can, at least not fruitfully. Org-brain uses headings as nodes, while Zetteldeft stays closer to the Zettelkasten idea of using short notes in separate files.

Just as a note: I really enjoyed Org-brain, which I used before starting with Zetteldeft. I wanted something a bit more flexible and with less reliance on Org-mode IDs, so that switching to different software (such as The Archive) would still be possible.

**Comment by michaelsjackson** [comment]
Org-brain can also use entries per org-file, actually this is how I started using it mostly, same as in TheBrain 3 there with .rtf files. obvz giving the nice visualizations, for far away view.

Yeah, I will watch a bit more of those youtube videos of zettelkasten.de members. I used zettelkasten in the past as well, then stopped using it after the java version (zkn 3).  I hope zetteldeft is powerful and complementary nicely.

org-brain + obvz + org-wiki + zetteldeft,  all on same org-files.  Why not?  
The trouble is with scalability, after 2000 .org files, for example.  But happily org-brain has now sub-brain feature in subdirectories.

For naming scheme of text files, I would prefer something like Karl Voit's technique, only extended a bit with Luhmann style branching, I would call it LID (Luhmann-ID).

Example:
2020-12-20-1223_LID_title_--_tag1_tag2.org
tags are for later entry points, exactly as Luhmann uses them as well, however to be able to use tags without having to dive into the .org file, directly from its name, it makes sense to add and remove them to the file name (append and remove, technique is from Karl Voit, he make extra programs for this purpose).

LID is required as well, also directly in the file name, not to have to dive into the file, so you can place a thought beside another one, in the hierarchy. It has to be also in the file name, for easy searchability later, via any kind of searching tools.  

All those great features one would get only by a structured file name. You could easily filter by date, tags, note position in hierarchy.

The zettlr guy, explains this and that in his youtube video, but I guess he did not get the point with the LID yet, he just ignores this fact, which is very important.  I would want to know beside which note I put another one, meaning those are related to each other just by the fact they are placed closely in the hierarchy tree.

**Comment by EFLS** [comment]
That's an interesting workflow, specifically tailored to your use-case.

I'm interested in the Luhmann style IDs since they convey hierarchy from the note's ID. For Luhmann himself this was especially important as his Zettelkasten was all paper. For a digital solution, I opted for simple time-based IDs as it makes things easier.

I'm not sure though whether it makes sense to simultaneously use Luhmann type IDs and time-based IDs. Would they not conflict?

In any case, I think implementing Luhmann IDs is beyond the current scope of Zetteldeft, but feel free to experiment and share your experiences :smile:

**Comment by michaelsjackson** [comment]
No they would not conflict, ID only needs to be unique, no matter what is its content, having the date and time as you did now, is great, we can filter by those later, just using file names, any search tool in any os would work, just perfect.  

However, if you leave out the Luhmann style branching in the id, like the zettlr guy did as well, then you are losing information, the hierarchical position of the note is lost, at least not available in file name any more. If the goal is building a long term huge or gigantic, lets call 100000 notes concept map (as Luhmann had around 90000 notes, in both of his Zettelkasten together) then placement of those notes in this gigantic hierarchy is very important.  How is zetteldeft placing notes in hierarchy? If I have note 1000 (out of 50000 for example), then want to know all notes close to that note, some before, some after, some around it, but not timewise of course, in terms of branching to and from it?  How would I do it in zetteldeft?  Still did not test zetteldeft yet.

My above notes is not any special workflow, just what you haved, extended with Karl Voith's naming method, which is most powerful and flexible across os, this was why he decided for this style I guess, plus adding the Luhmann style branching, but directly also into the file name.  Of course all those file name operations should happen via special tools or functions for them, not doing manually. It has to be easy, quick and comfortable.

**Comment by EFLS** [comment]
> Or asked in a different way, how and where is branching information saved in zetteldeft, if anywhere?

It isn't, at least not from a note's filename. Or rather: that is what the user should do him- or herself by leaving marks in the notes. You can create a note and a link to it at once, thus "branching" from the current note. But this isn't indicated from the filename.

There are many ways to create structure: manually maintaining "structure notes" or "hubs" (as also suggested by the guys at zettelkasten.de), working with tags, using the semi-automated functions provided by Zetteldeft (with `zetteldeft-insert-list-links`), etc.

**Comment by michaelsjackson** [comment]
I have seen those structure notes videos, it is a great idea, but you can use it additionally, not as the only possible way.  If adding normal links is the only thing, then you can use also org-wiki, the good old wiki.  It is just the same, here only each entry or wiki page or thought or zettel has the current date as its title, so you are saving yourself typing any name there.  Hmm, not sure what is the added value of zetteldeft yet.

There is also org-roam, with its backlinks per entry, I should list it above as well.  And good old wikis had also these backlinks, not sure why those are celebrated now as something genius?

It seems to me, currently what comes closest to Luhmann style organizing is something like org-brain at minimum, as each entry has its place somewhere in a huge map, you can go to any point there, and continue from there, visiting all the neighbours while traveling, if you want. Of course there you can also create extra structure notes, with a collection of links to various other entries.

What is clear so far, it is about placement of entries in the huge map first (1).  Second linking options for achieving various gains (2).

As org-brain uses friend,parent,child links, hopefully in future also input,output links for connecting or inserting or placement of entries in the huge map, point (1) from above.  You go to this map point first, then hit p, c, f which creates those connections to that point automatically.

In a similar way, if you would want to branch in the file name, you can say, c, branching of into child direction, an entry 1000, would get 1000a, like in Luhmanns case, switching or toggling between numbers and letters would show a switch of braching. If you would want to continue, in 1 year from 1000a, but without wanting to make that one bigger (Luhmann had to do so, because of limited paper space), you can add 1000a1, 1000a2 ...
Or add different alternative branches, then we need to go one step back first to 1000, then adding as 1000b, parallel to 1000a.

What is missing is this main thing, go somewhere, then branch off from there.  Normal links should not be mixed with such placement position branches.  Luhmann was very carefully placing his zettel/paper in a precise position, why did he do it?  Because each precise position is a fixed position in this huge map, he can go to any such position, finding related thoughts there, what is coming as a sequence next from there, or as a branching from there.  Just he can reproduce later, how those thoughts were build up, he does not need to rething those connections again, they are now available in hardware fixed form, as a big sculpture in of, he can close his eyes and follow the sculpture, until he reaches some end point, where he needs to think again, or has a new idea somewhere for a new continuation or branching again.  And if this part is missing, I guess a very big part of Zettelkasten technique is not there, as far as I see it.  Luhmann found such a simple and worhing technique for giving IDs, containing its branching information, why not using same, has only advantages, no disadvantages.  Finding parallel branches would be very easy later, and this is the whole point, what can you do with your notes later, not now.

I should stop now, I hope some of my points got a bit more clear.

**Comment by michaelsjackson** [comment]
Here the video to Karl Voits system.
https://www.youtube.com/watch?v=rckSVmYCH90

**Comment by EFLS** [comment]
I'm somewhat familiar with Voit's approach. In any case, I'd encourage you to take this discussion to the forums over at zettelkasten.de, where there is a lot of exchange going on about different approaches to digital Zettelkasten.
